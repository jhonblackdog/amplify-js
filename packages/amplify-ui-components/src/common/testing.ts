export const pixelThreshold = 10;
