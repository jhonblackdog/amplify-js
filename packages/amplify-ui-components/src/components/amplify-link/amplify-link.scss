.link {
  color: #ff9900;
  cursor: pointer;
  &:link {
    color: var(--amplify-amazon-orange);
    text-decoration: none;
  }

  &:hover {
    color: var(--amplify-light-amazon-orange);
    text-decoration: underline;
  }

  &:active {
    color: var(--amplify-button-click);
    text-decoration: underline;
  }
}
