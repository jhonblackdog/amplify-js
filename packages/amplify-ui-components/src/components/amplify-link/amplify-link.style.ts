import { css } from 'emotion';

export const link = css`
  color: #ff9900;
  cursor: pointer;
  :link {
    color: var(--amazon-orange);
    text-decoration: none;
  }

  :hover {
    color: var(--light-amazon-orange);
    text-decoration: underline;
  }
  
  :active {
    color: var(--button-click);
    text-decoration: underline;
  }
`;
