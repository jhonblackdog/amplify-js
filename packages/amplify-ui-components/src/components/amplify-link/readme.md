# amplify-link



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description | Type      | Default        |
| --------------- | ---------------- | ----------- | --------- | -------------- |
| `overrideStyle` | `override-style` |             | `boolean` | `false`        |
| `role`          | `role`           |             | `string`  | `'navigation'` |


## Dependencies

### Used by

 - [amplify-examples](../amplify-examples)
 - [amplify-sign-in](../amplify-sign-in)

### Graph
```mermaid
graph TD;
  amplify-examples --> amplify-link
  amplify-sign-in --> amplify-link
  style amplify-link fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
