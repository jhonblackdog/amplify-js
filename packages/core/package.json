{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
