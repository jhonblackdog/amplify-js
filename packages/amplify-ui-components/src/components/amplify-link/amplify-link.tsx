import { Component, Element, Prop, h } from '@stencil/core';
import { link } from './amplify-link.style';
import { styleNuker } from '../../common/helpers';
import { AMPLIFY_UI_PREFIX } from '../../common/constants';

const STATIC_LINK_CLASS_NAME = `${AMPLIFY_UI_PREFIX}--link`;

@Component({
  tag: 'amplify-link',
  shadow: false,
})
export class AmplifyLink {
  @Element() el: HTMLElement;

  @Prop() role: string = 'navigation';
  @Prop() overrideStyle: boolean = false;

  render() {
    return (
      <a class={styleNuker(this.overrideStyle, STATIC_LINK_CLASS_NAME, link)} role={this.role}>
        <slot />
      </a>
    );
  }
}
