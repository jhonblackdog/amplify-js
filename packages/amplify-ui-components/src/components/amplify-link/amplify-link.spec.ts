import { newSpecPage } from '@stencil/core/testing';
import { AmplifyLink } from './amplify-link';

describe('amplify-link spec:', () => {
  describe('Component logic ->', () => {
    let link;

    beforeEach(() => {
      link = new AmplifyLink();
    });

    it('should render false for style override prop by default', () => {
      expect(link.role).toEqual('navigation');
    });
    
    it('should render false for style override prop by default', () => {
      expect(link.overrideStyle).toBe(false);
    });
  });
  describe('Render logic ->', () => {
    it('should render a link by default', async () => {
      const page = await newSpecPage({
        components: [AmplifyLink],
        html: `<amplify-link></amplify-link>`
      });

      expect(page.root).toMatchSnapshot();
    });

    it('should render only class `amplify-ui--link` when override style is set to true', async () => {
      const page = await newSpecPage({
        components: [AmplifyLink],
        html: `<amplify-link override-style="true"></amplify-link>`
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});