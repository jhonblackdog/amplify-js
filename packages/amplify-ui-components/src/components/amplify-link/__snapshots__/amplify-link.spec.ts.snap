// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-link spec: Render logic -> should render a link by default 1`] = `
<amplify-link>
  <!---->
  <a class="amplify-ui--link css-3x35s3" role="navigation"></a>
</amplify-link>
`;

exports[`amplify-link spec: Render logic -> should render only class \`amplify-ui--link\` when override style is set to true 1`] = `
<amplify-link override-style="true">
  <!---->
  <a class="amplify-ui--link" role="navigation"></a>
</amplify-link>
`;
